hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DataStreamer.java

```diff
@@ -1188,6 +1198,43 @@ public void run() {
             }
           }
 
+          if (slowNodesFromAck.isEmpty()) {
+            if (!slowNodeMap.isEmpty()) {
+              slowNodeMap.clear();
+            }
+          } else {
+            Map<DatanodeInfo, Integer> newSlowNodeMap = new HashMap<>();
+            for (DatanodeInfo slowNode : slowNodesFromAck) {
+              if (!slowNodeMap.containsKey(slowNode)) {
+                newSlowNodeMap.put(slowNode, 1);
+              } else {
+                int oldCount = slowNodeMap.get(slowNode);
+                newSlowNodeMap.put(slowNode, ++oldCount);
+              }
+            }
+            slowNodeMap = newSlowNodeMap;
```

Comment:
Maybe it doesn't need to create new HashMap.
```suggestion
            for (DatanodeInfo slowNode : slowNodesFromAck) {
              if (!slowNodeMap.containsKey(slowNode)) {
                slowNodeMap.put(slowNode, 1);
              } else {
                int oldCount = slowNodeMap.get(slowNode);
                slowNodeMap.put(slowNode, ++oldCount);
              }
            }
```

Reply:
@tasanuma Sure, this version is modified to make it more readable.
I will share another implementation later. Hope you can check later.

Reply:
@tasanuma Could you help to check the implementation in https://github.com/symious/hadoop/commit/973af5e7a959f86f8b8807472a063c60193a2cdc ?

Reply:
Sorry, I didn't realize that `slowNodeMap` has discontinuous nodes, so my suggestion is wrong.

> Could you help to check the implementation in https://github.com/symious/hadoop/commit/973af5e7a959f86f8b8807472a063c60193a2cdc ?

It seems more readable than the previous one. Please use it. Thanks.